### PR TITLE
MAT-106 listen for charge success event + new chargeId DB field

### PR DESCRIPTION
**Pull request by tbg-jasonfung**
- MAT-106 Add new table field chargeId and related getter and setter methods
- MAT-106 refactor and listen for charge succeeded hook + start storing charge Id
- Update Stripe test data with charge object structure + rename files
- Update unit tests

### CODE REVIEW

File: src/Application/Actions/Hooks/StripeUpdate.php [hunk 1]
```diff
@@ -78,12 +77,20 @@ protected function action(): Response
...
+
+        if (!$donation) {
+            throw new DomainRecordNotFoundException('Donation not found');
+        }
+        
         // For now we support the happy success path,
         // as this is the only event type we're handling right now,
         // convert status to the one SF uses.
         if ($event->data->object->status === 'succeeded') {
+            $donation->setChargeId($event->data->object->id);
```
Comment (tbg-jasonfung) [inline src/Application/Actions/Hooks/StripeUpdate.php:93]: @NoelLH do you think it's worth sending the charge Id to SF and storing it there as well, or you think it'll be cleaner to just have payment intent Id to identify a donation on SF side?

Comment (NoelLH) [reply src/Application/Actions/Hooks/StripeUpdate.php:93]: I don't think it really needs to be in SF too @tbg-jasonfung as long as MatchBot can link them up. We can always check the object associations in Stripe admin.

File: src/Migrations/Version20200826135513.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Add field for Stripe charge ID
+ */
+final class Version20200826135513 extends AbstractMigration
+{
+    public function getDescription() : string
+    {
+        return '';
```
Comment (NoelLH) [inline src/Migrations/Version20200826135513.php:17]: @tbg-jasonfung it's good to set this too so console commands show the purpose.

File: tests/Application/Actions/Hooks/StripeUpdateTest.php [hunk 3]
```diff
@@ -24,16 +25,10 @@ public function testUnsupportedAction(): void
         $container = $app->getContainer();
 
         $body = file_get_contents(dirname(__DIR__, 3) . '/TestData/StripeWebhook/pi_canceled.json');
-        $donation = $this->getTestDonation();
         $webhookSecret = $container->get('settings')['stripe']['webhookSecret'];
         $time = (string) time();
 
         $donationRepoProphecy = $this->prophesize(DonationRepository::class);
-        $donationRepoProphecy
```
Comment (NoelLH) [inline tests/Application/Actions/Hooks/StripeUpdateTest.php:32]: We can probably remove this prophecy completely if it has no mocked methods left?

File: src/Application/Actions/Hooks/StripeUpdate.php [hunk 4]
```diff
@@ -78,18 +77,26 @@ protected function action(): Response
...
     }
 
-    public function handlePaymentIntentSucceeded(Event $event, $donation): Response
+    public function handleChargeSucceeded(Event $event): Response
     {
+        /** @var Donation $donation */
+        $donation = $this->donationRepository->findOneBy(['transactionId' => $event->data->object->payment_intent]);
+
+        if (!$donation) {
+            throw new DomainRecordNotFoundException('Donation not found');
```
Comment (NoelLH) [inline src/Application/Actions/Hooks/StripeUpdate.php:86]: This should probably info log and return a 204 as well so as not to break multi-sandbox use

**Review by NoelLH — CHANGES_REQUESTED** [review]
Key stuff looking good @tbg-jasonfung, just a few small suggestions

File: tests/Application/Actions/Hooks/StripeUpdateTest.php (hunk 3 above)
Comment (tbg-jasonfung) [reply tests/Application/Actions/Hooks/StripeUpdateTest.php:32]: @NoelLH When I remove this prophecy I get the following error:

`Error: Call to a member function setCampaignRepository() on null`

Comment (NoelLH) [reply tests/Application/Actions/Hooks/StripeUpdateTest.php:32]: Ah, OK @tbg-jasonfung. If you look at the trace of that call, it's to do with the 'real' `DonationRepository` setup's reliance on `EntityManagerInterface` in `repositories.php`. It looks like we do need to mock the donation repo in this case but not the EM, so we can remove that and it's DI override instead.

**Review by NoelLH — APPROVED** [review]

**Pull request merged**